src/implementation.js

```diff
@@ -0,0 +1,129 @@
+function _foldToObject(memo, [k, v]) {
+  memo[k] = v;
+  return memo;
+}
+
+function _objectMap(obj, fn) {
+  return _entries(obj).map(fn).reduce(_foldToObject, {});
+}
+
+function _entries(obj) {
+  let stringKeys = Object.getOwnPropertyNames(obj);
+  let symKeys = Object.getOwnPropertySymbols(obj);
+  return [...stringKeys, ...symKeys].map(p => [p, obj[p]]);
+}
+
+
+export default class Protocol extends null {
+  constructor({
+    name: protocolName,
+    extends: _extends = [],
+    requires = {},
+    staticRequires = {},
+    provides = {},
+    staticProvides = {},
+  } = {}) {
+    super();
+    this._name = protocolName;
+    this._extends = _extends;
+    this._requires = _objectMap(requires, ([name, sym]) => [name, sym == null ? this._createSymbol(name, { value: sym }) : sym]);
+    this._staticRequires = _objectMap(staticRequires, ([name, sym]) => [name, sym == null ? this._createSymbol(name, { value: sym }) : sym]);
+    this._provides = provides;
+    this._staticProvides = staticProvides;
+    // TODO: check that there's no conflicts between requires/staticRequires/provides/staticProvides
+    Object.assign(
```

Comment:
you also may want to use a babel transform for this to https://npmjs.com/object.assign, like https://github.com/airbnb/react-dates/blob/master/.babelrc#L7 (that way it will work down to ES5; even if someone's shammed Symbol)